File: w10/d04/Aziz_Hasanov/jungle_tree/app/controllers/monkeys_controller.rb
class MonkeysController < ApplicationController
  def index
    monkeys = Monkey.all
    render json: monkeys.to_json
  end
  def create
    monkeys = Monkey.create(monkey_params)
    render json: monkeys.to_json
  end
  def show
    monkey = Monkey.find(params[:id])
    render json: monkey.to_json
  end
  def update
    monkey = Monkey.find(params[:id])
    monkey.update(monkey_params)
    render json: monkeys.to_json
  end
  def destroy
    Monkey.delete(params[:id])
    render json: monkeys.to_json
  end

  private

  def monkey_params
    params.require(:monkey).permit(:name)
  end
end

File: w11/d01/Paul_Gasbarra/node_bananas/monkeysController.js
var Monkey = require(path.join(__dirname, 'monkey'));

exports.index = function(req, res){
  Monkey.find(function(err, monkeys){
    if (err) return console.log(err);
    res.send(monkeys);
  })
}

exports.create = function(req, res){
  Monkey.create({ name: req.body.name }, function(err, monkey){
    if (err) return console.log(err);
    res.send(monkey);
  })
}

exports.update = function(req, res){
  Monkey.findById(req.params.id, function(err, monkey){
    if (err) return console.log(err);
    monkey.update({ name: req.body.name },
      function(err, monkey){
      if (err) return console.log(err);
      return res.send({'success' : 'ok'});
      res.send(monkey);
  });

  })
}

exports.delete = function(req, res){
  Monkey.findById(req.params.id, function(err, monkey){
    if (err) return console.log(err);
    monkey.remove(function( err){
      if (err) return console.error(err);
      return res.send({'success' : 'ok'});
    });
    res.send(monkey);

  })
}
